src/exodus/test/sigmadb_tests.cpp

```diff
@@ -348,4 +375,64 @@ BOOST_AUTO_TEST_CASE(get_anonimity_group_by_iterator)
     BOOST_CHECK(pubs == std::vector<exodus::SigmaPublicKey>(anonimityGroupList.begin(), anonimityGroupList.end()));
 }
 
-BOOST_AUTO_TEST_SUITE_END()
\ No newline at end of file
+BOOST_AUTO_TEST_CASE(group_size_default)
+{
+    std::string testingDefaultFile = "MP_txlist_groupsize_test";
+
+    auto CMPMintListProxy = GetCMPMintList(testingDefaultFile, 0);
+    uint16_t defaultGroupSize = exodus::CMPMintList::MAX_GROUP_SIZE;
+    BOOST_CHECK_EQUAL(CMPMintListProxy->GetGroupSize(), defaultGroupSize);
+
+    CMPMintListProxy->Clear();
+}
+
+BOOST_AUTO_TEST_CASE(group_size_customsize)
+{
+    std::string testingDefaultFile = "MP_txlist_groupsize_test";
+
+    auto CMPMintListProxy = GetCMPMintList(testingDefaultFile, 120);
+    BOOST_CHECK_EQUAL(CMPMintListProxy->GetGroupSize(), 120);
+
+    CMPMintListProxy->Clear();
+}
+
+BOOST_AUTO_TEST_CASE(group_size_exceed_limit)
+{
+    std::string testingDefaultFile = "MP_txlist_groupsize_test";
+
+    uint16_t defaultGroupSize = exodus::CMPMintList::MAX_GROUP_SIZE;
+
+    CMPMintList* CMPMintListProxy = nullptr;
+    BOOST_CHECK_EXCEPTION(
+        CMPMintListProxy = GetCMPMintList(testingDefaultFile, defaultGroupSize + 1),
+        std::invalid_argument,
+        [] (const std::invalid_argument& e) {
+            return std::string("group size exceed limit") == e.what();
+        }
+    );
+
+    if (CMPMintListProxy) {
+        CMPMintListProxy->Clear();
+    }
+}
+
+BOOST_AUTO_TEST_CASE(use_differnet_group_size_from_database)
+{
+    std::string testingDefaultFile = "MP_txlist_groupsize_test";
+
+    CMPMintList* CMPMintListProxy = GetCMPMintList(testingDefaultFile, 10);
+    BOOST_CHECK_EXCEPTION(
+        CMPMintListProxy->InitGroupSize(11),
```

Comment:
We should open and close then open again with different group size instead.

Reply:
fixed